tests/integration/tiles/tiles.spec.ts

```diff
@@ -46,80 +46,140 @@ describe('tiles', function () {
 
-        expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
-        expect(response.body).toHaveProperty('message', 'minZoom must be less than or equal to maxZoom');
-        expect(response).toSatisfyApiSpec();
+          const response = await requestSender.postTilesByBboxRequest(bbox, 0, 1);
+          expect(response.status).toBe(httpStatusCodes.OK);
+
+          const response2 = await requestSender.postTilesByBboxRequest(bbox, 0, 1);
+
+          expect(response2.status).toBe(httpStatusCodes.CONFLICT);
+          expect(response2.body).toHaveProperty('message', 'Request already in queue');
+          expect(response2).toSatisfyApiSpec();
+        });
       });
 
-      it('should return 409 if the request is already in queue', async function () {
-        const bbox = getBbox();
+      describe('POST /tiles/list', function () {
+        it('should return 400 if the body is not a list', async function () {
+          const response = await requestSender.postTilesList({} as []);
+
+          expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
+          expect(response.body).toHaveProperty('message', 'request.body should be array');
+          expect(response).toSatisfyApiSpec();
+        });
+
+        it('should return 400 if the body is empty', async function () {
+          const response = await requestSender.postTilesList([]);
+
+          expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
+          expect(response.body).toHaveProperty('message', 'request.body should NOT have fewer than 1 items');
+          expect(response).toSatisfyApiSpec();
+        });
 
-        const response = await requestSender.postTilesByBboxRequest(bbox, 0, 1);
-        expect(response.status).toBe(httpStatusCodes.OK);
+        it('should return 400 if the body is not a valid tile', async function () {
+          const response = await requestSender.postTilesList([{ x: 0, y: 0 } as Tile]);
 
-        const response2 = await requestSender.postTilesByBboxRequest(bbox, 0, 1);
+          expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
+          expect(response.body).toHaveProperty('message', "request.body[0] should have required property 'z'");
+          expect(response).toSatisfyApiSpec();
+        });
 
-        expect(response2.status).toBe(httpStatusCodes.CONFLICT);
-        expect(response2.body).toHaveProperty('message', 'Request already in queue');
-        expect(response2).toSatisfyApiSpec();
+        it('should return 400 if the body contains tile that is out of bounds', async function () {
+          const response = await requestSender.postTilesList([{ z: 0, x: 1, y: 1, metatile: 8 } as Tile]);
+
+          expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
+          expect(response.body).toHaveProperty('message', 'x index out of range of tile grid');
+          expect(response).toSatisfyApiSpec();
+        });
       });
     });
     describe('Sad Path', function () {
```

Comment:
add new line between happy and bad and bad and sad

Reply:
done